### PR TITLE
Ui/pricing metric sparkle

### PR DESCRIPTION
**Pull request by chelshaw**
This PR adds some more safeguards against pricing metrics date inputs, as well as capabilities checking

Config tab is hidden when a policy denies access to metrics config, such as: 
```hcl
path "sys/internal/counters/*" {
    capabilities = ["read"]
}

path "sys/internal/counters/config" {
    capabilities = ["deny"]
}
```
policy-no-config-update

Config edit button is hidden when a policy allows only read access to metrics config, such as: 
```hcl
path "sys/internal/counters/*" {
    capabilities = ["read"]
}
```
policy-no-config-update

Error message appears when the start date is father than the retained months setting
start-date-error

Error message appears when the end date is the current month or later
end-date-error

### CODE REVIEW

File: ui/app/templates/vault/cluster/metrics/index.hbs [hunk 1]
```diff
@@ -21,18 +21,20 @@
           Vault usage
         {{/link-to}}
       {{/link-to}}
-      {{#link-to
-        "vault.cluster.metrics.config"
-        tagName="li"
-        activeClass="is-active"
-      }}
+      {{#if model.config.configPath.canRead}}
         {{#link-to
```
Comment (Monkeychip) [inline ui/app/templates/vault/cluster/metrics/index.hbs:25]: link-to within another link-to, is that right, or am I reading it wrong?

**Review by Monkeychip — APPROVED** [review]
Nice work!

File: ui/app/templates/vault/cluster/metrics/index.hbs (hunk 1 above)
Comment (chelshaw) [reply ui/app/templates/vault/cluster/metrics/index.hbs:25]: This was copypasta but you're right, it doesn't look like it needs to have both nested 🤔

Comment (chelshaw) [reply ui/app/templates/vault/cluster/metrics/index.hbs:25]: Just confirmed, this is how the tabs are in `secret-list-header` component, and removing the inner one causes strange styling issues. Gonna leave it for now, but worth looking into in the future

**Pull request merged**